### PR TITLE
add init_daemon_lock_file()

### PR DESCRIPTION
**Pull request by cgzones**
needed for ntp

### CODE REVIEW

**Comment by cgzones** [comment]
any comments on this one?

**Comment by pebenito** [comment]
What are examples where this would be used?

**Comment by cgzones** [comment]
the init script for ntp on debian:
https://sources.debian.net/src/ntp/1:4.2.8p9%2Bdfsg-2.1/debian/ntp.init/#L30

File: policy/modules/system/init.if [hunk 1]
```diff
@@ -512,6 +512,39 @@ interface(`init_daemon_pid_file',`
...
+##	</summary>
+## </param>
+#
+interface(`init_daemon_lock_file',`
+	gen_require(`
+		attribute daemonpidfile;
+		type initrc_t;
+	')
+
+	typeattribute $1 daemonpidfile; # only needed for init_t permissions
```
Comment (pebenito) [inline policy/modules/system/init.if]: I'd prefer not to overload the `daemonpidfile`. Put the related permissions here, and then I'll merge this. Should there be more callers, we can consider a new attribute, similar to `daemonpidfile`.

**Comment by cgzones** [comment]
avoided usage of daemonpidfile

**Pull request merged**